addons/mrp/wizard/mrp_product_produce.py

```diff
@@ -175,5 +176,25 @@ class MrpProductProduceLine(models.TransientModel):
 
     @api.onchange('lot_id')
     def _onchange_lot_id(self):
+        res = {}
         if self.product_id.tracking == 'serial':
             self.qty_done = 1
+            lots = set([])
+            # Check duplication of the serial number.
+            for move in self.product_produce_id.produce_line_ids.filtered(lambda l: l.lot_id) - self:
```

Comment:
i would prefer a collections.Counter here too. but if you want to keep it this way it's ok too.